tests/integration/auth.spec.coffee

```diff
@@ -30,8 +30,10 @@ describe 'SDK authentication', ->
 		describe 'resin.auth.authenticate()', ->
 
 			it 'should eventually be a valid token given valid credentials', ->
-				resin.auth.authenticate(credentials).then(resin.token.isValid).then (isValid) ->
-					m.chai.expect(isValid).to.be.true
+				resin.auth.authenticate(credentials)
+				.then(resin.auth.loginWithToken)
+				.then(resin.auth.getKey).then (key) ->
+					m.chai.expect(key).to.be.a('string')
```

Comment:
We definitely need some new tests here too. At the very least we need tests that log in with an API key, and checks the basic methods do sensible things.